src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -119,13 +127,25 @@ private void DisconnectTagComputer()
 
         private (SolutionServices solutionServices, IClassificationService classificationService)? TryGetClassificationService(ITextSnapshot snapshot)
         {
-            if (_workspace?.Services.SolutionServices is not { } solutionServices)
-                return null;
+            var workspace = _workspace;
+            var contentType = snapshot.ContentType;
+            var lastCachedServices = _lastCachedServices;
```

Comment:
>_lastCachedServices [](http://example.com/codeflow?start=37&length=19)

Should this be cleared out on DisconnectFromWorkspace?

Otherwise, this code would no longer detect being called after that and return the cached service.

Reply:
Makes sense